File: src/gui/Licence.java
package gui;

import java.awt.BorderLayout;
import java.awt.Insets;
import java.io.InputStream;
import java.util.Scanner;

import javax.swing.JFrame;
import javax.swing.JScrollPane;
import javax.swing.JTextArea;

public class Licence
{
    private JFrame licenceFrame;

    public Licence()
    {
        final InputStream istream = getClass().getResourceAsStream("licence-gpl.txt");

        @SuppressWarnings("resource")
        final String licenseText = new Scanner(istream, "UTF-8").useDelimiter("\\A").next();

        this.licenceFrame = new JFrame();
        getFrame().setTitle("Licence");
        getFrame().setBounds(100, 100, 640, 800);

        final JTextArea textField = new JTextArea();
        textField.setEditable(false);
        textField.setMargin(new Insets(10, 10, 10, 10));
        textField.setAlignmentX(0);
        getFrame().getContentPane().add(textField, BorderLayout.CENTER);
        textField.setText(licenseText);
        textField.setCaretPosition(0);

        final JScrollPane scroll =
            new JScrollPane(
                textField,
                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);

        getFrame().getContentPane().add(scroll);
        getFrame().setVisible(true);
    }

    private JFrame getFrame()
    {
        return this.licenceFrame;
    }
}

File: src/gui/Licence.scala
package gui

import java.awt.{BorderLayout, Insets}
import java.io.InputStream
import java.util.Scanner
import javax.swing.{JFrame, JScrollPane, JTextArea, ScrollPaneConstants}

object Licence extends JFrame {
    val istream: InputStream = getClass.getResourceAsStream("licence-gpl.txt")
    val licenseText: String = new Scanner(istream, "UTF-8").useDelimiter("\\A").next

    this.setTitle("Licence")
    this.setBounds(100, 100, 640, 800)

    val textField: JTextArea = new JTextArea
    textField.setEditable(false)
    textField.setMargin(new Insets(10, 10, 10, 10))
    textField.setAlignmentX(0)
    textField.setText(licenseText)
    textField.setCaretPosition(0)

    this.add(textField, BorderLayout.CENTER)

    val scroll: JScrollPane = new JScrollPane(textField,
        ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS, ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER)
    this.getContentPane.add(scroll)

    def showLicense(): Unit = this.setVisible(true)
}
